core/src/main/java/org/apache/carbondata/hadoop/CarbonInputSplit.java

```diff
@@ -419,7 +417,7 @@ public Segment getSegment() {
     } else {
       out.writeInt(0);
     }
-    out.writeBoolean(writeDeleteDelta);
+    writeDeleteDeltaFile(out);
```

Comment:
Better write boolean instead of int, it can save 3 bytes.

Reply:
done